controller/users_blackbox_test.go

```diff
@@ -804,6 +804,53 @@ func (s *TestUsersSuite) TestListUsersByEmailOKEmptyResult() {
 	require.Len(s.T(), result.Data, 0)
 }
 
+func (s *TestUsersSuite) TestHideEmailOK() {
+	boolTrue := true
+
+	// given user1
+	user1 := s.createRandomUser("TestListUsersOK1")
+	identity := s.createRandomIdentity(user1, account.KeycloakIDP)
+
+	// when
+	email := user1.Email
+
+	// by default, email is public.
+	_, result := test.ListUsersOK(s.T(), nil, nil, s.controller, &email, nil, nil, nil)
+	returnedUser := result.Data[0].Attributes
+	require.Equal(s.T(), email, *returnedUser.Email)
+	require.False(s.T(), *returnedUser.EmailPrivate)
+
+	secureService, secureController := s.SecuredController(identity)
+
+	contextInformation := map[string]interface{}{
+		"last_visited": "yesterday",
+		"space":        "3d6dab8d-f204-42e8-ab29-cdb1c93130ad",
+		"rate":         100.00,
+		"count":        3,
+	}
+	updateUsersPayload := createUpdateUsersPayload(nil, nil, nil, nil, nil, nil, nil, nil, contextInformation)
+	updateUsersPayload.Data.Attributes.EmailPrivate = &boolTrue
+	_, updateResult := test.UpdateUsersOK(s.T(), secureService.Context, secureService, secureController, updateUsersPayload)
+
+	// Email will be visible to the one who it belongs to
+	require.True(s.T(), *updateResult.Data.Attributes.EmailPrivate)
+	require.Equal(s.T(), user1.Email, *updateResult.Data.Attributes.Email)
+
+	// But when you try to access the same with an API which doesn't respect auth,
+	// it wouldn't be visible.
+	_, result = test.ListUsersOK(s.T(), nil, nil, s.controller, &email, nil, nil, nil)
+	returnedUserResult := result.Data[0]
+	require.Equal(s.T(), "", *returnedUserResult.Attributes.Email)
+
+	// even though the email_private=true,
+	// the email address is visible to the user if her user token is passed.
```

Comment:
The comment and test do not match. This is not what you are testing here. You are testing that api/users/:identityID doesn't return private emails. We should keep this test but fix the comment and add another test to the user controller test to actually test what your describe in the comment :)

Reply:
Agreed, looks like the /api/user and the /api/users confused me.
Thanks!

6583a918d37a389ab2aba36e546e01c89965f419